### ISSUE TITLE
How to debug false positives?

### ISSUE THREAD

**Issue by kkmuffme** [vimeo/psalm#2604]
I am getting an error by psalm for PossiblyUndefinedArrayOffset for code that is essentially this (literally, I just copied this from my file & put it in psalm with renamed variables. Just excluded previous & later code from my app to get a small snippet)

https://psalm.dev/r/88b792583f

```php
<?php
  
$array = array( 'foo' => 'hello', 'bar' => 'world' );

$key = 'bar';
$exists = false;
if ( isset( $array[ $key ] ) ) {
  $exists = true;
}

if ( $exists === true ) {
  $xyz = $array[ $key ];
}
```

Obviously psalm.dev doesn't report an error here, since this is a false positive I get by psalm in my application only.
What steps can I take to help debug the issue why psalm thinks within my application this could be PossiblyUndefinedArrayOffset?

**Comment by muglug** [comment]
You're probably running into this: https://psalm.dev/r/e9eb16a845

Because you have code that checks for `isset` on that entry, Psalm now thinks it's possibly undefined.

I'd suggest inlining the `isset` check: https://psalm.dev/r/f4f8b79b93

**Comment by kkmuffme** [comment]
Yes exactly. But it is possible undefined, which is why I do the isset check.

I can't inline it, since in reality there are 50 lines of other code between that, I just simplified it for the example.
Just ignored the error now
